app/controllers/pieces_controller.rb

```diff
@@ -19,29 +19,35 @@ def checks_before_move
     flashmessages
     destination_x = piece_params[:column_coordinate].to_i
     destination_y = piece_params[:row_coordinate].to_i
+    current_row = @piece.row_coordinate
+    current_column = @piece.column_coordinate
     render plain: 'Invalid Move', status: :bad_request if @piece.not_moved?(destination_x, destination_y)
     if !@piece.obstructed?(destination_x, destination_y) && @piece.valid_move?(destination_x, destination_y) && check_turn == @piece.color
       change_turn
       @piece.move_to!(destination_x, destination_y)
       current_game.set_game_status
-      flashmessages
-      redirect_to game_path(@piece.game_id)
+      Pusher.trigger('my-channel2', 'piece_moved', message: flashmessages,
```

Comment:
Currently - if multiple games are going on, this will push an update to every game that's open (not necessarily the correct game).

What you can do to fix that, is have the channel name be dynamic. Could be something like `"game-#{ game.id }"` so that it only updates that specific game page that is open